src/main/java/duke/other/Storage.java

```diff
@@ -0,0 +1,152 @@
+ */
+public class Storage {
+    protected String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Returns an ArrayList of Tasks loaded from the file specified by the file path.
+     *
+     * @return An ArrayList of Tasks
+     * @throws FileNotFoundException If the file is not found in the file path
+     */
+    public ArrayList<Task> load() throws FileNotFoundException {
+        Path path = Paths.get(filePath);
+        File noteFile = new File(path.toString());
+        if (Files.notExists(path)) {
+            noteFile = new File("data/duke.txt");
+        }
+        return fileToArray(noteFile.toString());
+    }
+
+    /**
+     * Returns an ArrayList of Tasks, after converting the Tasks in the file into the corresponding Tasks objects and
+     * adding them into an ArrayList.
+     *
+     * @param filePath file path of the input file of Tasks
+     * @return ArrayList of Tasks containing all the Tasks from the input file
+     * @throws FileNotFoundException If the file is not found in the file path
+     */
+    private static ArrayList<Task> fileToArray(String filePath) throws FileNotFoundException {
+        File f = new File(filePath);
+        Scanner sc = new Scanner(f);
+        ArrayList<Task> taskList = new ArrayList<>();
+        while (sc.hasNext()) {
+            String note = sc.nextLine();
+            String[] noteArr = note.split(">");
+            String noteType = noteArr[0];
+            boolean taskIsDone = (noteArr[1]).equals("true") ? true : false;
+            String taskDesc = noteArr[2];
+
+            if (noteType.equals("T")) {
+                taskList.add(new Todo(taskDesc, taskIsDone));
+            } else if (noteType.equals("E")) {
+                if (noteArr.length == 5) {
+                    taskList.add(new Event(taskDesc, LocalDate.parse(noteArr[3]), LocalTime.parse(noteArr[4]),
+                            taskIsDone));
+                } else if (noteArr.length == 4) {
+                    taskList.add(new Event(taskDesc, LocalDate.parse(noteArr[3]), taskIsDone));
+                }
+            } else {
+                if (noteArr.length == 5) {
+                    taskList.add(new Deadline(taskDesc, LocalDate.parse(noteArr[3]), LocalTime.parse(noteArr[4]),
+                            taskIsDone));
+                } else if (noteArr.length == 4) {
+                    taskList.add(new Event(taskDesc, LocalDate.parse(noteArr[3]), taskIsDone));
+                }
+            }
+        }
+        return taskList;
+    }
+
+    /**
+     * Returns a String to be stored in the file specified by the file path, after converting the Tasks objects into
+     * Strings.
+     *
+     * @param tasks TaskList of all the Tasks
+     * @return String of all Tasks, in the format for storage
+     */
+    public static String stringifyArray(TaskList tasks) {
```

Comment:
this function is a little obscure! might wanna consider using some top magic to remove all the duplicated portions!